### DISCUSSION TITLE
My images are sometimes converted to "colormap" instead of RGB

### DISCUSSION THREAD

**Discussion by stiofanoriain** [ImageMagick/ImageMagick#3969] [Help]
Hi all

I'm not sure why this is happening as it only happens *sometimes*.

image.png is as follows:

`$ file image.png` 
`image.png: PNG image data, 192 x 192, 8-bit/color RGBA, non-interlaced`

I want to convert it to RGB. I use the following command:

`convert image.png -flatten +matte image.png`

This sometimes works (converts it to `PNG image data, 192 x 192, 8-bit/color RGB, non-interlaced`) but for some reason it randomly becomes `PNG image data, 192 x 192, 8-bit **colormap**, non-interlaced`.

How do I force it to become RGB every time?

Thank you!

**Comment by snibgo** [comment]
By default, IM uses a PNG encoding method that minimizes the size of the file. This doesn't change the data.

For example, if your image has less than 256 different colours, a paletted ("colormap") file can be used, and that will take less disk space than non-paletted.

If, for some reason, your workflow can't use paletted files, use a "PNGn:" prefix on the output filename, sch as "PNG00:" or "PNG48:" or whaever. See https://www.imagemagick.org/script/formats.php#supported
